### PR TITLE
Add Deneb beacon API types

### PR DESCRIPTION
**Pull request by henridf**
- Introduce the `ProduceBlockResponseV2` type for handling responses to `GET /eth/v2/validator/blocks/{slot}` (https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Validator/produceBlockV2)

- Introduce the `RestPublishedSignedBlockContents` type for handling the request body to `POST /eth/v1/beacon/blocks` (https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/ValidatorRequiredApi/publishBlock)

And a few other small changes towards Deneb block production support.

### CODE REVIEW

File: beacon_chain/spec/eth2_apis/rest_types.nim [hunk 1]
```diff
@@ -723,6 +749,20 @@ type
 func `==`*(a, b: RestValidatorIndex): bool =
   uint64(a) == uint64(b)
 
+func init*(T: type ForkedSignedBeaconBlock,
+           contents: RestPublishedSignedBlockContents): T =
+  case contents.kind
```
Comment (tersec) [inline beacon_chain/spec/eth2_apis/rest_types.nim]: This case statement can be replaced with a case expression by removing all the `return`s, which matches the suggested use case in https://status-im.github.io/nim-style-guide/language.result.html#practical-notes for expression returns.

Comment (henridf) [reply beacon_chain/spec/eth2_apis/rest_types.nim]: done in c69b5878cf48b1f8e5f4100fae4077906f1b4b0b

**Comment by henridf** [comment]
The jenkins error looks unrelated:

```
sh: 1: cannot create /home/jenkins/workspace/2_platforms_linux_x86_64_PR-5060@tmp/durable-78769bdf/jenkins-result.txt.tmp: Directory nonexistent

mv: cannot stat '/home/jenkins/workspace/2_platforms_linux_x86_64_PR-5060@tmp/durable-78769bdf/jenkins-result.txt.tmp': No such file or directory

missing workspace /home/jenkins/workspace/2_platforms_linux_x86_64_PR-5060 on linux-06
```

**Comment by tersec** [comment]
> The jenkins error looks unrelated:
> 
> ```
> sh: 1: cannot create /home/jenkins/workspace/2_platforms_linux_x86_64_PR-5060@tmp/durable-78769bdf/jenkins-result.txt.tmp: Directory nonexistent
> 
> mv: cannot stat '/home/jenkins/workspace/2_platforms_linux_x86_64_PR-5060@tmp/durable-78769bdf/jenkins-result.txt.tmp': No such file or directory
> 
> missing workspace /home/jenkins/workspace/2_platforms_linux_x86_64_PR-5060 on linux-06
> ```

Agree, rerunning.

**Pull request merged**